thirdparty/build-thirdparty.sh

```diff
@@ -1210,6 +1210,16 @@ build_libdeflate() {
 
 #clucene
 build_clucene() {
+    if [[ -z ${USE_AVX2} ]]; then
```

Comment:
check if clucene can be enhanced to build avx2 and non-avx2 version co-exist in the library, and determine which one to use during the runtime where the cpuset info can be leveraged.

Reply:
use a common flag for the whole script